src/pages/index.js

```diff
@@ -112,22 +181,77 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModalProfile);
+  setButtonText(getSubmitButton(evt), true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModalProfile);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(getSubmitButton(evt), false);
+    });
 }
 
 function handleAddPostSubmit(evt) {
   evt.preventDefault();
+  const form = evt.target;
   const inputValues = {
-    name: addPostCaptionInput.value,
-    link: addPostLinkInput.value,
+    name: form.elements["add-post-caption-input"].value,
+    link: form.elements["add-post-link-input"].value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  closeModal(addPostModal);
-  disabledButton(addPostSubmitButton, settings);
+  setButtonText(getSubmitButton(evt), true);
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      closeModal(addPostModal);
+      disabledButton(addPostSubmitButton, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      setButtonText(getSubmitButton(evt), false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(getSubmitButton(evt), true);
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      avatar.src = data.avatar;
+      closeModal(avatarModal);
+      avatarModalForm.reset();
```

Comment:
* Please, disable the `avatar` submit button after submission as well https://snipboard.io/1fQPlr.jpg 


```js

      avatarModalForm.reset();
      disabledButton(getSubmitButton(evt), settings);
```